### PR TITLE
Dev 2646 IDV Activity Chart Tooltips

### PR DESCRIPTION
**Pull request by jonhill13**
**High level description:**

Add tooltips to IDV Activity Chart Bars

**JIRA Ticket:**
[DEV-2646](https://federal-spending-transparency.atlassian.net/browse/DEV-2646)

**Mockup:**
https://bahdigital.invisionapp.com/share/ACIAA6A938E#/screens/296010428

The following are ALL required for the PR to be merged:
- [x] Code review
- [x] Design review @AGuyNamedMarco 
- [x] Verified cross-browser compatibility

### CODE REVIEW

File: src/js/components/awardv2/idv/activity/ActivityChartTooltip.jsx [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+                transform: ''
+            },
+            windowWidth: 0,
+            windowHeight: 0,
+            awardingAgency: props.data.awardingAgencyName,
+            recipient: props.data.recipientName,
+            truncated: false
+        };
+
+        this.measureWindow = this.measureWindow.bind(this);
```
Comment (MaxwellKendall) [inline src/js/components/awardv2/idv/activity/ActivityChartTooltip.jsx]: Should we debounce/throttle this fn?

File: src/js/components/awardv2/idv/activity/ActivityChartTooltip.jsx [hunk 2]
```diff
@@ -0,0 +1,301 @@
+/**
+ * FederalAccountsTreeTooltip.jsx
+ * Created by Jonathan Hill 5/2/19
+ */
```
Comment (MaxwellKendall) [inline src/js/components/awardv2/idv/activity/ActivityChartTooltip.jsx]: update this?

File: src/js/components/awardv2/idv/activity/ActivityChartTooltip.jsx [hunk 3]
```diff
@@ -80,8 +80,8 @@ export default class IdvActivityTooltip extends React.Component {
     }
 
     truncateText(awardingAgencyNameDiv, recipientNameDiv, theTooltipWidth) {
-        const { graphWidth, recipientName, awardingAgencyName } = this.props.data;
-        if (this.props.data.graphWidth < theTooltipWidth) {
+        return new Promise((resolve) => {
```
Comment (MaxwellKendall) [inline src/js/components/awardv2/idv/activity/ActivityChartTooltip.jsx]: Tested this locally and it appears we don't need to return a promise, we can just use the async / await on line 159. `setState` itself is async so I don't think we need to wrap all this in a promise. Maybe I'm missing something but that's my understanding.

Comment (jonhill13) [reply src/js/components/awardv2/idv/activity/ActivityChartTooltip.jsx]: Not sure we really want to do that considering https://stackoverflow.com/questions/47019199/why-does-async-await-work-with-react-setstate

**Pull request merged**